### ISSUE TITLE
Have you tried Cloud Hypervisor?

### ISSUE THREAD

**Issue by sboeuf** [hocus-dev/hocus#118]
## ❓ Questions/Help/Support

I've just read your [blogpost](https://hocus.dev/blog/qemu-vs-firecracker/) about the move from Firecracker to QEMU and the reasons why Firecracker wasn't fitting your needs. Have you ever considered using [Cloud Hypervisor](https://github.com/cloud-hypervisor/cloud-hypervisor)? It's been designed for the use cases you describe, and doesn't have any of the limitation that made you move away from Firecracker. And in comparison with QEMU, the codebase is much smaller and written in Rust. It actually relies on multiple Rust-VMM crates which are used from Firecracker as well.

**Comment by hugodutka** [comment]
Hi Sebastian, we did consider it. Here's a short writeup that we did about why we can't use it yet: https://github.com/hocus-dev/hocus/blob/main/rfd/0002-workspace-runtime.md. Is there any specific reason why you're asking?

**Comment by sboeuf** [comment]
Hi! I've been working on Cloud Hypervisor for several years and it looked like a great fit for your use case.
Ok I see the only blocker was the `discard` feature for storage, right? I assume you're using a block device, which means the only thing that's missing is to implement the `VIRTIO_BLK_F_DISCARD` feature for the `virtio-block` device. I'm not actively working on the project, but that's definitely something you could ask the community to implement for you (unless you want to do it).

**Comment by hugodutka** [comment]
Yes, the main problem was the lack of the discard feature. However, we've already integrated with QEMU and it works great, so it's not a pressing issue for us right now. We wouldn't see any immediate benefits from switching to Cloud Hypervisor at this time.
